Some builders

rubendm92 opened the pull request:
### Summary

I've implemented builders for effect state.set and coeffects state.get and global.get. I haven't done anymore because I wanted feedback if this the way we'll like to expose the builders.

You can see some usages in the react todos example.

trikitrok commented on examples/with-react-todos/src/todos/TodoList/events.js:
```diff
@@ -28,18 +29,14 @@ export default function registerTodoListEvents() {
-    return {
-      'state.set': { todos: todos },
-    };
+    return stateBuilder.set({ todos: todos });
```
I'd prefer to read `state.set({ todos: todos });`

trikitrok commented on examples/with-react-todos/src/todos/TodoList/events.js:
```diff
@@ -12,7 +13,7 @@ export default function registerTodoListEvents() {
       };
     },
-    [coeffect('global.get', 'apiUrl')]
+    [globalBuilder.get('apiUrl')]
```
I'd prefer to read `global.get('apiUrl')`

rubendm92 replied:
Problem here is that `global` is already a declared variable in Node, so I can not call it like that :(

trikitrok commented on examples/with-react-todos/src/todos/TodoList/events.js:
```diff
@@ -61,15 +58,15 @@ export default function registerTodoListEvents() {
       };
     },
-    [coeffect('state.get', { todos: 'todos' })]
+    [stateBuilder.get({ todos: 'todos' })]
```
I'd prefer to read `state.get({ todos: 'todos' })`

trikitrok replied:
That's true :(

rubendm92 commented:
> It's really great.
> 
> I added some comments regarding the naming (it happens in more places thatn the ones I commented).
> 
> The idea behind those comments is that I'd rather keep using the coeffects & effects names directly instead of coeffectBuilder & effectBuilder because in my opinion it's shorter and reads better (also looking at the way these ones are used, they seem more like factories than builders).
> 
> Example:
> Using `state.set({ todos: todos });` instead of `stateBuilder.set({ todos: todos });`
> 
> Another option might be using a factory `stateSet({ todos: todos });` which I think is enough for these simple coeffects & effects.
> 
> I think the need for builders will come when creating more complicated coeffects or effects, such as the http effects that have many parameters of which some are optional.

Thanks for the feedback!

I also prefer the `state.set({ todos: todos })` format, so I changed it to that. I had to name the `global.get` coeffect as `globalVariable.get` since I can not use `global` as a name. If you find a better name I'm open to suggestions :)

trikitrok commented:
`globalVariable.get` reads great!!
